bot/services/delete_message_service.py

```diff
@@ -0,0 +1,62 @@
+import logging
+import typing as t
+
+import discord
+
+from bot.data.message_repository import MessageRepository
+from bot.services.base_service import BaseService
+from bot.messaging.events import Events
+from bot.messaging import messenger
+
+log = logging.getLogger(__name__)
+
+
+class DeleteMessageService(BaseService):
+    """
+    This service allows for messages sent by the bot to be deleted
+    The messags by default are allowed by deleted by admins and the person who called the bot
+    """
+
+    def __init__(self, *, bot):
+        super().__init__(bot)
+        self.messages = {}
+
+    # Called When a cog would like to be able to delete a message or messages
+    @BaseService.Listener(Events.on_set_deletable)
+    async def set_message_deletable(self, messagesToDelete: t.List[discord.Message],
+                                    roles: t.List[str] = [],
```

Comment:
the roles should be a `discord.Role` not an str, this will allow us to do id comparisons on the objects. 

so `t.List[discord.Role]` this will require some minor refactoring with how you detect that a user has a role